### ISSUE TITLE
Integrate p7zip into busybox-w32?

### ISSUE THREAD

**Issue by tn1997tn** [rmyorston/busybox-w32#293]
What do you think about it? I think p7zip is useful to have.

**Comment by rmyorston** [comment]
I think there's a lot of overlap between what p7zip provides and what busybox-w32 already has.

Which features of p7zip would it make sense to add to busybox-w32? I guess support for .7z files, possibly .cab.

I also think upstream BusyBox wouldn't be interested in such features.

**Comment by tn1997tn** [comment]
> I think there's a lot of overlap between what p7zip provides and what busybox-w32 already has.
> 
> Which features of p7zip would it make sense to add to busybox-w32? I guess support for .7z files, possibly .cab.
> 
> I also think upstream BusyBox wouldn't be interested in such features.

BusyBox can't extract .7z archive. p7zip could extract a lot more than it own .7z format, including proprietary format like .rar.

**Comment by ale5000-git** [comment]
I don't think it would really need to support everything but still I think there are 2 useful formats to support decompression: 7z and zst

**Comment by yetanothergeek** [comment]
> I think there are 2 useful formats to support decompression: 7z and zst

For what it's worth, there were patches by two different developers submitted to upstream busybox to add zstd support. Neither of them was accepted, but you should still be able find the patches and the discussions on the mailing list.

**Comment by rmyorston** [comment]
zstd is definitely an upstream issue. There were BusyBox mailing list discussions in September and December 2021.

There's also this perspective from the zstd side: [https://github.com/facebook/zstd/issues/2806](https://github.com/facebook/zstd/issues/2806)

**Comment by naksyl** [comment]
Windows builtin tar (libarchive's bsdtar) `$WINDIR/System32/tar` can handle 7zip archives - not sure what compression is used by deafult. (tested on Windows 10 build 19045.3448). Listing and extracting autodetect 7zip format.

Windows 11 `KB5030310` update (22621.2361) (not sure if Windows 10 gets equivalent) adds support for additional formats to builtin tar (3.6.2) - compiled with:
- zlib/1.2.5.f-ipp
- liblzma/5.2.5
- bz2lib/1.0.8
- libzstd/1.5.4

Example of creation 7zip archive:
``` sh
$WINDIR/System32/tar -cf busybox-w32.7z --format 7zip \
                     --options "7zip:compression=lzma2,7zip:compression-level=9" \
                     busybox-w32
```

One can put `alias bsdtar="$WINDIR/System32/tar.exe"` to ~/.profile

Usage lacks for documentation of 7zip so here are debian manpages:
- [bsdtar(1)](https://manpages.debian.org/unstable/libarchive-tools/bsdtar.1.en.html)
- [archive_write_set_filter_option(3)](https://manpages.debian.org/unstable/libarchive-dev/archive_write_set_options.3.en.html) - find options for archive creation here
